googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1147,6 +1147,46 @@ class NativeArray {
     GTEST_CONCAT_TOKEN_(gtest_label_testthrow_, __LINE__): \
       fail(gtest_msg.value)
 
+#define GTEST_TEST_THROW_WHAT_(statement, expected_exception, regex, fail) \
+  GTEST_AMBIGUOUS_ELSE_BLOCKER_ \
+  if (::testing::internal::ConstCharPtr gtest_msg = "") { \
+    bool gtest_caught_expected = false; \
+    try { \
+      GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_(statement); \
+    } \
+    catch (expected_exception const& e) { \
+      const ::testing::internal::RE& gtest_regex = (regex); \
+      const bool matched = ::testing::internal::RE::PartialMatch(::testing::PrintToString(e), gtest_regex); \
```

Comment:
Seems odd to make the macro part of gmock just to get matchers. There aren't any comparable macros in gmock. Why not just leave out support for matchers until they are unified with gtest? Even without matchers, it would provide similar functionality to the EXPECT_FATAL_FAILURE macro.